bin/ttab

```diff
@@ -289,17 +301,26 @@ CMD_ACTIVATE='if not frontmost then
 end if'
 
     # For use with -g and -G: commands for saving and restoring the previous application
+CMD_KEEP_TAB_OPEN="$SHELL"
```

Comment:
`-g` currently behaves like `-G` should:

 * `-g` is only for not activating the terminal program itself - inside the program, the new tab / window should become active

* `-G` also should not active the terminal program itself, but is _additionally_ for retaining the active tab and window in the terminal program). 

In AppleScript these behaviors are implemented by letting the unwanted activation happen and then switching the focus back - not sure if we can do that in Gnome Terminal. 

We may have to limit ourselves to not supporting `-g` at all, and to support `-G` only  terminal-internally, and there only for _tabs_, by omitting `--active` (_windows_ activate invariably, seemingly, even without `--active`).

Reply:
I now see the conceptual confusion: `-g` and `-G` were never meant to run programs _invisibly_ - they still create tabs / windows, but do not _activate_ the terminal application / the newly crated tab /window.

The logic of using a shell to keep the new tab/window open is currently the reverse:  a stay-open shell is used, inside of which the custom commands run; to _replace_ the shell with a new process the custom command must be passed with `exec` - that is, _not_ keeping the tab / window open is an _opt-in_, via the custom command.

Ideally, we'd do the same in Gnome Terminal, but that appears not to be an option; your technique of appending  `; $SHELL` is the next best thing, it seems, but it is suboptimal in that it doesn't preserve the environment of the shell in which the custom command executed.

* PowerShell actually does support both executing via `-c` while keeping the shell open via `-noexit`, so if we check for `pwsh` as the leaf component of `$SHELL`, we could do add `-noexit` instead of  using `$CMD_KEEP_TAB_OPEN`.

* Otherwise, for slightly improved efficiency ,I suggest:

```suggestion
CMD_KEEP_TAB_OPEN="exec $SHELL"
```

Reply:
Yes, it's a little confusing for me -g -G options description. Not sure I understand on what affects suggested change, but I can apply it.